aten/src/ATen/native/mps/UnaryConstants.h

```diff
@@ -35,4 +35,18 @@ kernel void erfinv_mps_kernel( device {0} *output [[buffer(0)]],
   }}
 
   output[index] = {0}(x);
-}})METAL";
+}}
+
+kernel void exp_kernel( device {0} *output [[buffer(0)]],
```

Comment:
Can you add tanh as well ?

Reply:
Sure, added, those complex implementation gets a bit hairy...